components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt

```diff
@@ -127,7 +210,66 @@ class DefaultAddonUpdater(
     internal fun getUniquePeriodicWorkName(addonId: String) =
         "$IDENTIFIER_PREFIX$addonId.periodicWork"
 
+    @VisibleForTesting
+    private fun getUniqueImmediateWorkName(extensionId: String) =
+        "$IDENTIFIER_PREFIX$extensionId.immediateWork"
+
+    private fun createNotification(ext: WebExtension) {
+        val channel = ChannelData(
+            NOTIFICATION_CHANNEL_ID, R.string.mozac_feature_addons_updater_notification_channel,
+            NotificationManagerCompat.IMPORTANCE_LOW
+        )
+        val channelId = ensureChannelExists(applicationContext, channel)
+        val intent =
+            applicationContext.packageManager.getLaunchIntentForPackage(applicationContext.packageName)?.apply {
+                flags = Intent.FLAG_ACTIVITY_NEW_TASK
+            } ?: throw IllegalStateException("Package has no launcher intent")
+
+        val allowIntent = Intent(applicationContext, NotificationReceiver::class.java).apply {
+            action = NOTIFICATION_ACTION_ALLOW
+            putExtra(NOTIFICATION_EXTRA_ADDON_ID, ext.id)
+        }
+        val allowPendingIntent: PendingIntent =
+            PendingIntent.getBroadcast(applicationContext, 0, allowIntent, 0)
+
+        val title =
+            applicationContext.getString(R.string.mozac_feature_addons_updater_notification_title)
+        val contentText = applicationContext.getString(
+            R.string.mozac_feature_addons_updater_notification_content,
+            "Ublock"
```

Comment:
Here I need a way to get an `add-on` from a `extension` to the the localize name.